onyo/lib/onyo.py

```diff
@@ -1066,9 +1100,13 @@ def set(self, paths: Iterable[Union[Path, str]], values: dict, dryrun: bool,
         if name_values:
             try:
                 self._update_names(assets_to_set, name_values)
-                self.clear_caches()
             except ValueError as e:
-                self.clear_caches()
+                self.clear_caches(assets=True,  # `onyo set` can rename assets
+                                  dirs=False,  # no szenario with `set`
+                                  files=True,  # in theory might change other YAML
```

Comment:
This is insufficiently specific. Vague "in theory" statements are not helpful.

This cache should be cleared because assets can be renamed, and assets are files.